### DISCUSSION TITLE
Should .where(drop=True) turn variables in an xarray Dataset from bool to float?

### DISCUSSION THREAD

**Discussion by rbavery** [pydata/xarray#5058] [Q&A] [answered]
I have a Dataset called `bayts` that contains a float variable and a boolean variable, both with dims (date, y, x). When I subset with `.where` to remove nan pixels:

`pixel_timeseries = bayts.isel(y=y, x=x).where(notnan_mask, drop=True)`

the boolean variable all of a sudden becomes a float, with True -> `1.` and False -> `0.` Should this happen or is this a bug?

**Comment by max-sixty** [comment, marked as answer]
This is currently expected behavior. 

It would be possible to retain `bool` with `drop=True`; I think there's an issue on this.
It's not possible to retain bool or int values without `drop=True`, since they don't have a missing value.

It's worth exploring whether you can use indexing for this.

**Reply by rbavery** [reply]
Thanks, I ended up using indexing. I had to convert the datetime index (used to make the notnan_mask) to a positional integer index so that I could use positional indexing to use views instead of array copies.
